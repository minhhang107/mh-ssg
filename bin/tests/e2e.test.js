const run = require("./run");

describe("end-to-end integration", () => {
  test("print help message when no arg ", async () => {
    const { stderr, exitCode } = await run();
    expect(exitCode).toBe(0);
    expect(stderr).toMatchSnapshot();
  });

  test("print help message when --help flag used", async () => {
    const { stderr, stdout, exitCode } = await run("--help");
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("print version when --version flag used", async () => {
    const { stderr, stdout, exitCode } = await run("--version");
    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("print error message when no input", async () => {
    const { stderr, exitCode } = await run("--input");
    expect(exitCode).toBe(1);
    expect(stderr).toMatchSnapshot();
  });

  test("print error message when input not found", async () => {
    const { stderr, exitCode } = await run("--input", "notfound.txt");
    expect(exitCode).toBe(1);
    expect(stderr).toMatchSnapshot();
  });

  test("print error message when input not supported", async () => {
    const { stderr, exitCode } = await run("--input", "sample.doc");
    expect(exitCode).toBe(1);
    expect(stderr).toMatchSnapshot();
  });
});
