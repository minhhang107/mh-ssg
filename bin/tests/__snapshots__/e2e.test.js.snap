// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`end-to-end integration print error message when input not found 1`] = `
"Usage: mh-ssg -i <filename> -o <outputpath> -s <stylesheet>

Options:
  -h, --help        Show help                                          [boolean]
  -i, --input       Specify input file or input folder (required)        [array]
  -o, --output      Specify output path                                  [array]
  -s, --stylesheet  Specify stylesheet for html file                    [string]
  -c, --config      Specify config file                                  [array]
  -v, --version     Show version number                                [boolean]
Input file does not exist. Please use a different file."
`;

exports[`end-to-end integration print error message when input not supported 1`] = `
"Usage: mh-ssg -i <filename> -o <outputpath> -s <stylesheet>

Options:
  -h, --help        Show help                                          [boolean]
  -i, --input       Specify input file or input folder (required)        [array]
  -o, --output      Specify output path                                  [array]
  -s, --stylesheet  Specify stylesheet for html file                    [string]
  -c, --config      Specify config file                                  [array]
  -v, --version     Show version number                                [boolean]
Input file does not exist. Please use a different file."
`;

exports[`end-to-end integration print error message when no input 1`] = `
"Usage: mh-ssg -i <filename> -o <outputpath> -s <stylesheet>

Options:
  -h, --help        Show help                                          [boolean]
  -i, --input       Specify input file or input folder (required)        [array]
  -o, --output      Specify output path                                  [array]
  -s, --stylesheet  Specify stylesheet for html file                    [string]
  -c, --config      Specify config file                                  [array]
  -v, --version     Show version number                                [boolean]
Input file does not exist. Please use a different file."
`;

exports[`end-to-end integration print help message when --help flag used 1`] = `
"Usage: mh-ssg -i <filename> -o <outputpath> -s <stylesheet>

Options:
  -h, --help        Show help                                          [boolean]
  -i, --input       Specify input file or input folder (required)        [array]
  -o, --output      Specify output path                                  [array]
  -s, --stylesheet  Specify stylesheet for html file                    [string]
  -c, --config      Specify config file                                  [array]
  -v, --version     Show version number                                [boolean]"
`;

exports[`end-to-end integration print help message when no arg  1`] = `
"Usage: mh-ssg -i <filename> -o <outputpath> -s <stylesheet>

Options:
  -h, --help        Show help                                          [boolean]
  -i, --input       Specify input file or input folder (required)        [array]
  -o, --output      Specify output path                                  [array]
  -s, --stylesheet  Specify stylesheet for html file                    [string]
  -c, --config      Specify config file                                  [array]
  -v, --version     Show version number                                [boolean]"
`;

exports[`end-to-end integration print version when --version flag used 1`] = `"You're running mh-ssg version 1.0.7"`;
